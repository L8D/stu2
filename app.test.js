/* globals describe, it, expect, jest */

require('dotenv').config()
const moment = require('moment-timezone')

const { handleHeartbeat } = require('./app')

describe('app', () => {
  it('should handle a heartbeat', async () => {
    const mockLogger = {
      log: jest.fn()
    }

    const mockRedisClient = {
      setAsync: jest.fn(),
      getAsync: jest.fn(() => null)
    }

    const mockTwilioClient = {
      messages: {
        create: jest.fn()
      }
    }

    await handleHeartbeat({
      nowMoment: moment('2018-11-28T23:18:56.525Z').tz('America/Chicago'),
      logger: mockLogger,
      redisClient: mockRedisClient,
      twilioClient: mockTwilioClient
    })

    expect({
      loggerCalls: mockLogger.log.mock.calls,
      twilioCalls: mockTwilioClient.messages.create.mock.calls,
      redisSetCalls: mockRedisClient.setAsync.mock.calls,
      redisGetCalls: mockRedisClient.getAsync.mock.calls
    }).toMatchSnapshot()
  })
})
