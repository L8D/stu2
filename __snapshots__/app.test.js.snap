// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should handle a heartbeat 1`] = `
Object {
  "loggerCalls": Array [
    Array [
      "handling action",
      Object {
        "deadline": "6:30pm",
        "sentAt": "2018-11-28T18:18:56-05:00",
        "type": "outbound",
        "value": "You need to eat dinner. You have 11 minutes.",
      },
    ],
    Array [
      "sending twilio message",
      Object {
        "body": "You need to eat dinner. You have 11 minutes.",
        "from": "+15125807478",
        "to": "+18324540603",
      },
    ],
    Array [
      "successfully sent twilio message",
      Object {
        "body": "You need to eat dinner. You have 11 minutes.",
        "from": "+15125807478",
        "to": "+18324540603",
      },
    ],
    Array [
      "handling action",
      Object {
        "key": "2018-11-28:tasks:17:announcement",
        "type": "set",
        "value": true,
      },
    ],
  ],
  "redisGetCalls": Array [
    Array [
      "2018-11-28:tasks:17:announcement",
    ],
  ],
  "redisSetCalls": Array [
    Array [
      "2018-11-28:tasks:17:announcement",
      true,
    ],
  ],
  "twilioCalls": Array [
    Array [
      Object {
        "body": "You need to eat dinner. You have 11 minutes.",
        "from": "+15125807478",
        "to": "+18324540603",
      },
    ],
  ],
}
`;
